### PR TITLE
Adds support for environment variable configuration

### PR DESCRIPTION
**Pull request by ForestEckhardt**
- Separates buildpack.yml parsing into its own structure to aide in
business logic when it comes to deprecating buildpack.yml

Please confirm the following:
* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have added an integration test, if necessary.

### CODE REVIEW

**Comment by ForestEckhardt** [comment]
TODO LIST:
* [ ] Figure out failing test cases.
* [ ] Add integration testing around environment variables having higher priority.
* [ ] Add logging for if a user is trying to configure the buildpack using environment variable.
* [ ] Write up documentation for the website as well as the README.

**Comment by wburningham** [comment]
Will using the `BP_GO_*` ENV variables clear the layers cache or will it suffer from the same issues reported in #88 and #89?

**Comment by ForestEckhardt** [comment]
@wburningham It will currently have those same issue but that is the next set of work we are looking onto around making sure that we rebuild the layers at an appropriate time.

**Review by ryanmoran — CHANGES_REQUESTED** [review]

File: go.mod [hunk 1]
```diff
@@ -5,6 +5,8 @@ go 1.14
 require (
 	github.com/BurntSushi/toml v0.3.1
 	github.com/buildkite/interpolate v0.0.0-20200526001904-07f35b4ae251
+	github.com/hashicorp/hcl v1.0.0
```
Comment (ryanmoran) [inline go.mod:8]:
```suggestion
```

**Review by ryanmoran — CHANGES_REQUESTED** [review]

**Review by ryanmoran — APPROVED** [review]

**Pull request merged**